### DISCUSSION TITLE
Why my SDK is looking for the credentials in the wrong place?

### DISCUSSION THREAD

**Discussion by angeloghiotto** [aws/aws-sdk-php#2670] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/2537009/228906963-a79a50cc-4029-4436-917e-cd24d24c548e.png)

I'm trying to use DynammoDB on Debian with PHP 8.2, and SDK is looking for credentials in /usr/sbin/.aws/credentials, not the home.

I've checked, and the PHP user has a .aws/credentials in the right place.

I can fix that using the following: 

`putenv('AWS_SHARED_CREDENTIALS_FILE=/path/to/your/credentials/file');`

But I would like to know why and how to fix that without using putenv or another workaround.

PS: I tried to set the credentials on the fly, but the DynammoDB SDK returns an error

**Comment by SamRemis** [comment, marked as answer]
Hi @angeloghiotto,
The main documentation on our credentials can be found here:
https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials_provider.html

There are a large number of ways you can authenticate, and at a quick glance it looks like you're trying to use the [process credential provider](https://github.com/aws/aws-sdk-php/blob/bb52cd91150647ec39fe01cc852a335df954e0b7/src/Credentials/CredentialProvider.php#L561) as part of our default chain.  If you look at the linked code, it defaults to looking for the credentials file in `self::getHomeDir() . '/.aws/credentials'`, where the [getHomeDir function](https://github.com/aws/aws-sdk-php/blob/bb52cd91150647ec39fe01cc852a335df954e0b7/src/Credentials/CredentialProvider.php#L717) where the home directory function looks for the `HOME` variable if it exists, and otherwise looks for `HOMEDRIVE` and `HOMEPATH`.

Since you're on Debian, it should be able to find the `HOME` variable and be looking in the right place.  Where are you expecting it to look?  To me, /usr/username/.aws/credentials seems like a valid place for that file to be.  The easiest work around would be putting the file into the directory where the SDK is looking for it.  

Alternatively, you could specifically pass in a directory to look in as shown in the example modified from the docs below:
```
$profile = 'production';
$path = '/full/path/to/credentials/file';

$provider = CredentialProvider::process($profile, $path);
$provider = CredentialProvider::memoize($provider);

$client = new DynamoDbClient([
    'region'      => 'us-west-2',
    'version'     => '2006-03-01',
    'credentials' => $provider
]);
```

**Reply by angeloghiotto** [reply]
Thanks for the reply; I later realized I was trying to use Polly credentials in Dynnamo, so I got the error using the credentials param.

The HOME issue is that my PHP was running as a user daemon; it was looking in the 'right' place for that user. I fixed it using the 'credentials' param as you said

[a comment by an automated account is not shown]
